src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -116,21 +133,43 @@
       :fx [[:dispatch [:navigate-to :wallet-transaction-progress]]]})))
 
 (defn- transaction-bridge
-  [{:keys [from-address to-address route]}]
-  (let [{:keys [from bridge-name amount-out gas-amount gas-fees]}           route
-        {:keys [gas-price max-fee-per-gas-medium max-priority-fee-per-gas]} gas-fees]
-    [{:BridgeName bridge-name
-      :ChainID    (:chain-id from)
-      :TransferTx {:From                 from-address
-                   :To                   to-address
-                   :Gas                  (money/to-hex gas-amount)
-                   :GasPrice             (money/to-hex (money/->wei :gwei gas-price))
-                   :Value                amount-out
-                   :Nonce                nil
-                   :MaxFeePerGas         (money/to-hex (money/->wei :gwei max-fee-per-gas-medium))
-                   :MaxPriorityFeePerGas (money/to-hex (money/->wei :gwei max-priority-fee-per-gas))
-                   :Input                ""
-                   :Data                 "0x"}}]))
+  [{:keys [from-address to-address token-id token-address route]}]
+  (let [{:keys [from bridge-name amount-out gas-amount
+                gas-fees]}                 route
+        eip-1559-enabled?                  (:eip-1559-enabled gas-fees)
+        {:keys [gas-price max-fee-per-gas-medium
+                max-priority-fee-per-gas]} gas-fees
+        transfer-tx                        (merge
+                                            (if eip-1559-enabled?
+                                              {:TxType               "0x02"
+                                               :MaxFeePerGas         (money/to-hex
+                                                                      (money/->wei
+                                                                       :gwei
+                                                                       max-fee-per-gas-medium))
+                                               :MaxPriorityFeePerGas (money/to-hex
+                                                                      (money/->wei
+                                                                       :gwei
+                                                                       max-priority-fee-per-gas))}
+                                              {:TxType   "0x00"
+                                               :GasPrice (money/to-hex (money/->wei :gwei
+                                                                                    gas-price))})
+                                            {:From  from-address
+                                             :To    (or token-address to-address)
+                                             :Gas   (money/to-hex gas-amount)
+                                             :Value amount-out
+                                             :Nonce nil
+                                             :Input ""
+                                             :Data  "0x"})]
```

Comment:
`cond->` could make the code more beautiful (and more performant than `merge`). WDYT?

```clojure
(cond-> {:From  from-address
         :To    (or token-address to-address)
         :Gas   (money/to-hex gas-amount)
         :Value amount-out
         :Nonce nil
         :Input ""
         :Data  "0x"})
  
  eip-1559-enabled?
  (assoc :TxType               "0x02"
         :MaxFeePerGas         (money/to-hex
                                (money/->wei
                                 :gwei
                                 max-fee-per-gas-medium))
         :MaxPriorityFeePerGas (money/to-hex
                                (money/->wei
                                 :gwei
                                 max-priority-fee-per-gas)))

  (not eip-1559-enabled?)
  (assoc :TxType   "0x00"
         :GasPrice (money/to-hex (money/->wei :gwei
                                              gas-price)))
```

Reply:
Sure! I will implement the suggestions you mentioned :)